org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchTests.java

```diff
@@ -4593,4 +4576,137 @@ public void testAnonymousTypeMethodReferenceJarSearchGh375() throws Exception {
+	String snippet1 = "package p;\n" +
+			"public class TestGh432 {\n" +
+			"  public static void main(String[] args) {\n" +
+			"    class R implements Runnable {\n" +
+			"      public void run() {\n" +
+			"        java.util.stream.Stream.empty().filter(TestGh432::missingReference).count();\n" +
+			"      }\n" +
+			"    }\n" +
+			"    new R().run();\n" +
+			"  }\n" +
+			"  public static boolean missingReference(Object arg) {\n" +
+			"    return true;\n" +
+			"  }\n" +
+			"}";
+	String snippet2 = "package p;\n" +
+			"public class TestGh432 {\n" +
+			"  private static void foo(Runnable r) {\n" +
+			"    r.run();\n" +
+			"  }\n" +
+			"  public static void main(String[] args) {\n" +
+			"    foo(new Runnable() {\n" +
+			"      public void run() {\n" +
+			"        java.util.stream.Stream.empty().filter(TestGh432::missingReference).count();\n" +
+			"      }\n" +
+			"    });\n" +
+			"  }\n" +
+			"  public static boolean missingReference(Object arg) {\n" +
+			"    return true;\n" +
+			"  }\n" +
+			"}";
+	String snippet3 = "package p;\n" +
+			"public class TestGh432 {\n" +
+			"  public static void main(String[] args) {\n" +
+			"    new Runnable() {\n" +
+			"      public void run() {\n" +
+			"        new Runnable() {\n" +
+			"          public void run() {\n" +
+			"            java.util.stream.Stream.empty().filter(TestGh432::missingReference).count();\n" +
+			"          }\n" +
+			"        }.run();\n" +
+			"      }\n" +
+			"    }.run();\n" +
+			"  }\n" +
+			"  public static boolean missingReference(Object arg) {\n" +
+			"    return true;\n" +
+			"  }\n" +
+			"}";
+	String snippet4 = "package p;\n" +
+			"public class TestGh432 {\n" +
+			"  private static void foo(Runnable r) {\n" +
+			"    r.run();\n" +
+			"  }\n" +
+			"  public static void main(String[] args) {\n" +
+			"    foo(() -> {\n" +
+			"      new Runnable() {\n" +
+			"        public void run() {\n" +
+			"          java.util.stream.Stream.empty().filter(TestGh432::missingReference).count();\n" +
+			"        }\n" +
+			"      };\n" +
+			"    });\n" +
+			"  }\n" +
+			"  public static boolean missingReference(Object arg) {\n" +
+			"    return true;\n" +
+			"  }\n" +
+			"}";
+	String[] testSnippets = {
+			snippet1,
+			snippet2,
+			snippet3,
+			snippet4,
+	};
+	String[] expectedMatches = {
+			"src/p/TestGh432.java void void p.TestGh432.main(String[]):R#1.run() [missingReference]",
+			"src/p/TestGh432.java void void p.TestGh432.main(String[]):<anonymous>#1.run() [missingReference]",
+			"src/p/TestGh432.java void void void p.TestGh432.main(String[]):<anonymous>#1.run():<anonymous>#1.run() [missingReference]",
+			"src/p/TestGh432.java \nJava Model Exception: Error in Java Model (code 969): run() {key=Lp/TestGh432$162;.run()V} [in <anonymous #1> [in run() [in <lambda #1> [in main(String[]) [in TestGh432 [in TestGh432.java [in p [in src [in gh432MethodReferencesSearchBug]]]]]]]]] does not exist",
```

Comment:
I don't understand the `JavaModelException` occurring here. I'm debugging to find out what is causing it. Stack trace of its creation is:

```
"main" #1 prio=5 os_prio=0 cpu=2537.33ms elapsed=41.73s tid=0x00007ffff0018000 nid=0x1183f at breakpoint [0x00007ffff7fc8000]
   java.lang.Thread.State: RUNNABLE
        at org.eclipse.jdt.core.JavaModelException.<init>(JavaModelException.java:75)
        at org.eclipse.jdt.internal.core.JavaElement.newNotPresentException(JavaElement.java:574)
        at org.eclipse.jdt.internal.core.JavaElement.openWhenClosed(JavaElement.java:617)
        at org.eclipse.jdt.internal.core.JavaElement.getElementInfo(JavaElement.java:328)
        at org.eclipse.jdt.internal.core.JavaElement.getElementInfo(JavaElement.java:314)
        at org.eclipse.jdt.internal.core.hierarchy.HierarchyResolver.remember(HierarchyResolver.java:441)
        at org.eclipse.jdt.internal.core.hierarchy.HierarchyResolver.rememberWithMemberTypes(HierarchyResolver.java:542)
        at org.eclipse.jdt.internal.core.hierarchy.HierarchyResolver.rememberAllTypes(HierarchyResolver.java:527)
        at org.eclipse.jdt.internal.core.hierarchy.HierarchyResolver.resolve(HierarchyResolver.java:874)
        at org.eclipse.jdt.internal.core.hierarchy.HierarchyResolver.resolve(HierarchyResolver.java:669)
        at org.eclipse.jdt.internal.core.hierarchy.HierarchyBuilder.buildSupertypes(HierarchyBuilder.java:129)
        at org.eclipse.jdt.internal.core.hierarchy.IndexBasedHierarchyBuilder.build(IndexBasedHierarchyBuilder.java:170)
        at org.eclipse.jdt.internal.core.hierarchy.TypeHierarchy.compute(TypeHierarchy.java:323)
        at org.eclipse.jdt.internal.core.hierarchy.TypeHierarchy.refresh(TypeHierarchy.java:1319)
        - locked <0x0000000111beb4a8> (a org.eclipse.jdt.internal.core.hierarchy.TypeHierarchy)
        at org.eclipse.jdt.internal.core.CreateTypeHierarchyOperation.executeOperation(CreateTypeHierarchyOperation.java:94)
        at org.eclipse.jdt.internal.core.JavaModelOperation.run(JavaModelOperation.java:740)
        at org.eclipse.jdt.internal.core.JavaModelOperation.runOperation(JavaModelOperation.java:806)
        at org.eclipse.jdt.internal.core.SourceType.newSupertypeHierarchy(SourceType.java:858)
        at org.eclipse.jdt.internal.core.SourceType.newSupertypeHierarchy(SourceType.java:809)
        at org.eclipse.jdt.internal.core.search.IndexSelector.getFocusedElementsAndTypes(IndexSelector.java:172)
        at org.eclipse.jdt.internal.core.search.IndexSelector.initializeIndexLocations(IndexSelector.java:233)
        at org.eclipse.jdt.internal.core.search.IndexSelector.getIndexLocations(IndexSelector.java:304)
        at org.eclipse.jdt.internal.core.search.JavaSearchParticipant.selectIndexURLs(JavaSearchParticipant.java:160)
        at org.eclipse.jdt.internal.core.search.PatternSearchJob.getIndexes(PatternSearchJob.java:186)
        at org.eclipse.jdt.internal.core.search.PatternSearchJob.ensureReadyToRun(PatternSearchJob.java:99)
        at org.eclipse.jdt.internal.core.search.processing.JobManager.performConcurrentJob(JobManager.java:220)
        at org.eclipse.jdt.internal.core.search.BasicSearchEngine.findMatches(BasicSearchEngine.java:235)
        at org.eclipse.jdt.internal.core.search.BasicSearchEngine.search(BasicSearchEngine.java:602)
        at org.eclipse.jdt.core.search.SearchEngine.search(SearchEngine.java:670)
        at org.eclipse.jdt.core.tests.model.JavaSearchTests.searchForMethodReferences(JavaSearchTests.java:4696)
        at org.eclipse.jdt.core.tests.model.JavaSearchTests.testAnonymousTypeMethodReferenceSearchGh432(JavaSearchTests.java:4680)
```

The exception occurs also when using Eclipse to compute the call hierarchy in this snippet...

Reply:
Though by the looks if an extra `JavaModelException` occurs in the result collector used for the test...